linera-chain/src/chain.rs

```diff
@@ -631,6 +631,17 @@ where
         let mut messages = Vec::new();
         let mut message_counts = Vec::new();
 
+        if *self.execution_state.system.closed.get() {
```

Comment:
Should it be in the tip state?

Reply:
I'd actually say no. The "closed" property now doesn't affect whether you can still add blocks and at which heights, it only restricts what can be inside the blocks. We might add more such restrictions later, and I don't think they should be part of the tip state.